File: src/MediCore.Desktop/lib/features/appointments/domain/entities/appointment.dart
class Appointment {
  final String id;
  final String patientId;
  final String doctorId;
  final DateTime appointmentDate;
  final String startTime;
  final String endTime;
  final String status;
  final String? chiefComplaint;
  final String? notes;
  final String? cancellationReason;
  final DateTime createdAt;

  const Appointment({
    required this.id,
    required this.patientId,
    required this.doctorId,
    required this.appointmentDate,
    required this.startTime,
    required this.endTime,
    required this.status,
    this.chiefComplaint,
    this.notes,
    this.cancellationReason,
    required this.createdAt,
  });
}

File: src/MediCore.API/MediCore.Domain/Entities/Medical/Appointment.cs
using MediCore.Domain.Common;
using MediCore.Domain.Enums;

namespace MediCore.Domain.Entities.Medical;

public class Appointment : BaseEntity
{
    public Guid TenantId { get; set; }
    public Guid BranchId { get; set; }
    public Guid PatientId { get; set; }
    public Guid DoctorId { get; set; }
    public DateOnly AppointmentDate { get; set; }
    public TimeOnly StartTime { get; set; }
    public TimeOnly EndTime { get; set; }
    public AppointmentStatus Status { get; set; } = AppointmentStatus.Booked;
    public string? ChiefComplaint { get; set; }
    public string? Notes { get; set; }
    public string? CancellationReason { get; set; }
    public DateTime? ArrivedAt { get; set; }
    public DateTime? StartedAt { get; set; }
    public DateTime? EndedAt { get; set; }
    public bool ReminderSent { get; set; } = false;

    // Navigation Properties
    public Patient Patient { get; set; } = null!;
    public Branch Branch { get; set; } = null!;
}